### ISSUE TITLE
[Bug]: BSR_DE seems not to work anymore

### ISSUE THREAD

**Issue by Detmud** [mampfes/hacs_waste_collection_schedule#1591]
### I Have A Problem With:

A specific source

### What's Your Problem

Source BSR_DE is not importing anything.

This is the Path to the Calendar
https://www.bsr.de/abfuhrkalender_ajax.php?script=dynamic_iCal_ajax&abf_strasse=Chaukenstr.,%2012524%20Berlin%20(Treptow-K%C3%B6penick)&abf_hausnr=3X&tab_control=Liste&abf_config_weihnachtsbaeume=&abf_config_restmuell=on&abf_config_biogut=on&abf_config_wertstoffe=on&abf_config_laubtonne=on

this is the Ical it should get.

BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//www.bsr.de//NONSGML kigkonsult.se iCalcreator 2.41.71//
CALSCALE:GREGORIAN
UID:65616338-3738-4064-a633-326437643934
BEGIN:VEVENT
UID:32386261-6536-4130-a166-356138626438
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Hausmüll
DTSTART;TZID=Europe/Berlin:20231228T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Hausmüll
END:VEVENT
BEGIN:VEVENT
UID:37383565-6638-4232-a331-633133633061
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Biogut
DTSTART;TZID=Europe/Berlin:20240102T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Biogut
END:VEVENT
BEGIN:VEVENT
UID:30626662-3731-4963-b836-653262613233
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Wertstoffe (Abholung durch ALBA)
DTSTART;TZID=Europe/Berlin:20240109T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Wertstoffe (Abholung durch ALBA)
END:VEVENT
BEGIN:VEVENT
UID:64356336-6132-4464-a339-646533373961
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Hausmüll
DTSTART;TZID=Europe/Berlin:20240110T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Hausmüll
END:VEVENT
BEGIN:VEVENT
UID:61636636-6661-4834-b063-653566636636
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Biogut
DTSTART;TZID=Europe/Berlin:20240115T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Biogut
END:VEVENT
BEGIN:VEVENT
UID:32326238-3862-4533-b361-353033303935
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Wertstoffe (Abholung durch ALBA)
DTSTART;TZID=Europe/Berlin:20240123T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Wertstoffe (Abholung durch ALBA)
END:VEVENT
BEGIN:VEVENT
UID:30396632-3239-4764-b439-336132306661
DTSTAMP:20231228T151659Z
DESCRIPTION:Abholung Hausmüll
DTSTART;TZID=Europe/Berlin:20240124T000000
LOCATION:Chaukenstr. 3X\,
SUMMARY:Abholung Hausmüll
END:VEVENT
END:VCALENDAR

### Source (if relevant)

BSR_DE

### Logs

```Shell
no relevant logs
```

### Relevant Configuration

```YAML
- name: bsr_de
      args:
        abf_strasse: Chaukenstr., 12524 Berlin (Treptow-Köpenick)
        abf_hausnr: 3*
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [x] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by PeterHoge** [comment]
Same here. 29.12.2023 was the last entry (Wertstoffe). All following are not shown. Tomorrow 30.12.2023 is the next (Biogut) (not shown).

**Comment by BrainInSight** [comment]
Same here. Seems that dates for 2024 are not downloaded

**Comment by 1aranzant** [comment]
Similar issue for Belgium, I guess we need to wait for 2024

**Comment by Detmud** [comment]
> Same here. 29.12.2023 was the last entry (Wertstoffe). All following in 2024 are not shown.

I dont have that. Can you share your Config ?

**Comment by 5ila5** [comment]
did some investigating on their website but I can't get a ICS file (which the source uses to get the dates) containing any collections for 2024 Even though there are some listed in the html table in List view. This could fix itself in 2024 but currently there is nothing I can do about it.

**Comment by 5ila5** [comment]
@1aranzant which source for Belgium do you mean? I might have a look at it (may prevent this for next year)

**Comment by 1aranzant** [comment]
@5ila5   Recycle!

**Comment by 5ila5** [comment]
@1aranzant 3 out of 4 test cases return collections for 2024 I guess some municipalities have not yet added their collection dates.
Do you see any dates entering your address on their website (https://www.recycleapp.be)?

**Comment by 1aranzant** [comment]
@5ila5 I do see the 2024 collection dates on the recycle website, but my HA calendar is empty :/

**Comment by tr4nt0r** [comment]
Seems you can only get a yearly ics calendar for the current year but if you select monthly you can already download ics files with dates for next year

**Comment by Detmud** [comment]
> Seems you can only get a yearly ics calendar for the current year but if you select monthly you can already download ics files with dates for next year


> did some investigating on their website but I can't get a ICS file (which the source uses to get the dates) containing any collections for 2024 Even though there are some listed in the html table in List view. This could fix itself in 2024 but currently there is nothing I can do about it.
> 
> edit: @Detmud how did you get this ical file containing events for 2024? I only got empty ICAL files or one with only 2023 events when testing


Sure.
Go to: https://www.bsr.de/abfuhrkalender-20520.php
Search for : Chaukenstr., 12524 Berlin (Treptow-Köpenick) and choose a Number.

Choose; Termine herrunterladen.
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/103764/512e99fe-d21e-4bde-a7cb-e02b91b62444)

Choose: Termindatei herunterladen
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/103764/24c4de76-bff7-49e9-8772-4b2d5a0791d7)

The URL should match my initial post but i change my Housenumber to 3x

**Comment by 5ila5** [comment]
@tr4nt0r added the fix in #1603 (but the source should hopefully work normally without this patch tomorrow)

This fix will be part of 1.45.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
